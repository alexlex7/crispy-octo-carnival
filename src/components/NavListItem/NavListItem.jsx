import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './NavListItem.module.css';
import PropTypes from 'prop-types';

export default function NavListItem({ to, text }) {
  return (
    <li>
      <NavLink
        to={to}
        className={({ isActive }) => (isActive ? styles.active : styles.item)}
      >
        {text}
      </NavLink>
    </li>
  );
}

NavListItem.propTypes = {
  to: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};
