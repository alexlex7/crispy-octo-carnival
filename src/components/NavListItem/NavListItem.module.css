.item {
  font-size: 16px;
  font-weight: 400;
  color: #161616;
  text-decoration: none;
}

.active {
  font-size: 16px;
  font-weight: 400;
  color: rgb(141, 233, 200);
}

.item:hover {
  color: #244d4d;
  text-decoration: underline;
}
