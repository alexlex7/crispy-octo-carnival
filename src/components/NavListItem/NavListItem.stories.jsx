import React from 'react';
import NavListItem from './NavListItem';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Navigation/NavListItem',
  component: NavListItem,
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <NavListItem {...args} />;

export const Default = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Default.args = {
  text: 'Home',
  to: '/',
};
