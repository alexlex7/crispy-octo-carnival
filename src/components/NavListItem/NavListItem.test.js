import { render, screen } from '@testing-library/react';
import NavListItem from './NavListItem';
import { BrowserRouter } from 'react-router-dom';

test('renders NavListItem component', () => {
  const { asFragment } = render(<NavListItem text="Home" to="/" />, {
    wrapper: BrowserRouter,
  });
  const linkItem = screen.getByRole('link', { name: 'Home' });
  expect(linkItem).toBeInTheDocument();
  expect(asFragment).toMatchSnapshot();
});
