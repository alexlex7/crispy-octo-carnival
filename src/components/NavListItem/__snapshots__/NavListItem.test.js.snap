// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavListItem component 1`] = `[Function]`;
